tests/functional/test_hydro_with_rulecurves.py

```diff
@@ -99,35 +108,38 @@
 
 
 def test_hydro_heuristic_monthly_part() -> None:
-    """ """
+    """Check that the cost of the monthly problem is the same in the POC and in Antares."""
     capacity = 1e07
-    intermonthly_breakdown = 1
-    folder_name = "hydro_with_rulecurves"
-    initial_level = 0.445 * capacity
 
-    initial_level, status, obj, _ = optimize_target(
-        inter_breakdown=intermonthly_breakdown,
-        folder_name=folder_name,
-        capacity=capacity,
-        scenario=0,
-        initial_level=initial_level,
-        horizon="monthly",
-        timesteps=list(range(12)),
-        total_target=None,
+    solving_output, monthly_output = optimize_target(
```

Comment:
With previous refacorization, we should see a call to the montly step function here